Change to urls API for unregistered URLs causes errors

<!-- DO NOT REMOVE THE HEADER -->
<!---HEADER START-->

<img height="20px" src="https://i.imgur.com/c7hUeb5.jpeg">

❌ **This issue is not open for contribution. Visit <a href="https://learningequality.org/contributing-to-our-open-code-base/" target="_blank">Contributing guidelines</a>** to learn about the contributing process and how to find suitable issues.

<img height="20px" src="https://i.imgur.com/c7hUeb5.jpeg">

<!---HEADER END-->

## Observed behavior
If the Facility plugin is disabled, the Device settings page does not load.

This is particularly problematic, as the facility page can be disabled from the same page!

## Errors and logs

```
8vue.runtime.common.prod.js:11 Error: URL pattern "kolibri:kolibri.plugins.facility:facility_management" not found
    at urlFunc (urls.js:76:15)
    at i.getFacilitySettingsPath (index.vue:794:1)
    at index.vue:665:45
    at i.de [as _l] (vue.runtime.common.prod.js:11:13326)
    at i.DeviceSettingsPagevue_type_template_id_814e494a_scoped_true_render (index.vue:657:29)
    at t._render (vue.runtime.common.prod.js:11:44429)
    at i.o (vue.runtime.common.prod.js:11:74040)
    at wn.value (vue.runtime.common.prod.js:11:24951)
    at wn.value (vue.runtime.common.prod.js:11:25594)
    at Vn (vue.runtime.common.prod.js:11:27459)
Ge @ vue.runtime.common.prod.js:11Entender este error
19vue.runtime.common.prod.js:11 Error: URL pattern "kolibri:kolibri.plugins.facility:facility_management" not found
    at urlFunc (urls.js:76:15)
    at i.getFacilitySettingsPath (index.vue:794:1)
    at index.vue:665:45
    at i.de [as _l] (vue.runtime.common.prod.js:11:13326)
    at i.DeviceSettingsPagevue_type_template_id_814e494a_scoped_true_render (index.vue:657:29)
    at t._render (vue.runtime.common.prod.js:11:44429)
    at i.o (vue.runtime.common.prod.js:11:74040)
    at wn.value (vue.runtime.common.prod.js:11:24951)
    at wn.value (vue.runtime.common.prod.js:11:25594)
    at Vn (vue.runtime.common.prod.js:11:27459)
Ge @ vue.runtime.common.prod.js:11
qe @ vue.runtime.common.prod.js:11
We @ vue.runtime.common.prod.js:11
t._render @ vue.runtime.common.prod.js:11
o @ vue.runtime.common.prod.js:11
value @ vue.runtime.common.prod.js:11
value @ vue.runtime.common.prod.js:11
Vn @ vue.runtime.common.prod.js:11
(anónimo) @ vue.runtime.common.prod.js:11
Ye @ vue.runtime.common.prod.js:11Entender este error
15device/#/settings:1 Uncaught (in promise) Error: A listener indicated an asynchronous response by returning true, but the message channel closed before a response was receivedEntender este error
2vue.runtime.common.prod.js:11 Error: URL pattern "kolibri:kolibri.plugins.facility:facility_management" not found
    at urlFunc (urls.js:76:15)
    at i.getFacilitySettingsPath (index.vue:794:1)
    at index.vue:665:45
    at i.de [as _l] (vue.runtime.common.prod.js:11:13326)
    at i.DeviceSettingsPagevue_type_template_id_814e494a_scoped_true_render (index.vue:657:29)
    at t._render (vue.runtime.common.prod.js:11:44429)
    at i.o (vue.runtime.common.prod.js:11:74040)
    at wn.value (vue.runtime.common.prod.js:11:24951)
    at wn.value (vue.runtime.common.prod.js:11:25594)
    at Vn (vue.runtime.common.prod.js:11:27459)
Ge @ vue.runtime.common.prod.js:11Entender este error
session.js:86 Error: URL pattern "kolibri:kolibri.plugins.facility:facility_management" not found
    at urlFunc (urls.js:76:15)
    at i.getFacilitySettingsPath (index.vue:794:1)
    at index.vue:665:45
    at i.de [as _l] (vue.runtime.common.prod.js:11:13326)
    at i.DeviceSettingsPagevue_type_template_id_814e494a_scoped_true_render (index.vue:657:29)
    at t._render (vue.runtime.common.prod.js:11:44429)
    at i.o (vue.runtime.common.prod.js:11:74040)
    at wn.value (vue.runtime.common.prod.js:11:24951)
    at wn.value (vue.runtime.common.prod.js:11:25594)
    at Vn (vue.runtime.common.prod.js:11:27459)
```

## Expected behavior
We should always guard cross-plugin URL references as other plugins may be disabled.

## User-facing consequences
If you disable the facility plugin, you have to use the command line to enable it.

## Steps to reproduce
Disable the facility plugin on the device settings page. When it reloads, the page will error.

--- a/kolibri/plugins/device/assets/src/views/DeviceSettingsPage/index.vue
+++ b/kolibri/plugins/device/assets/src/views/DeviceSettingsPage/index.vue
@@ -289,7 +289,7 @@
       </section>
 
       <!-- List of separate links to Facility Settings pages -->
-      <section v-if="isMultiFacilitySuperuser">
+      <section v-if="isMultiFacilitySuperuser && getFacilitySettingsPath()">
         <h2>{{ $tr('configureFacilitySettingsHeader') }}</h2>
         <ul class="ul-reset">
           <template>

--- a/packages/kolibri/__tests__/urls.spec.js
+++ b/packages/kolibri/__tests__/urls.spec.js
@@ -136,7 +136,7 @@ describe('UrlResolver', () => {
 
   describe('Error handling', () => {
     test('throws error for non-existent patterns', () => {
-      expect(() => Urls.nonExistentPattern()).toThrow('URL pattern "nonExistentPattern" not found');
+      expect(Urls.nonExistentPattern).toBeUndefined();
     });
 
     test('handles initialization with no plugin data', () => {
@@ -144,9 +144,7 @@ describe('UrlResolver', () => {
       plugin_data.urls = undefined;
       const UrlsNoData = createUrlResolver();
 
-      expect(() => UrlsNoData['user_profile_detail'](123)).toThrow(
-        'URL pattern "user_profile_detail" not found',
-      );
+      expect(UrlsNoData['user_profile_detail']).toBeUndefined;
     });
     test('throws error if URL pattern contains a dash', () => {
       jest.resetModules();

--- a/packages/kolibri/urls.js
+++ b/packages/kolibri/urls.js
@@ -65,19 +65,16 @@ class UrlResolver {
 
     const patterns = this._patterns[name];
     if (!patterns) {
-      urlFunc = () => {
-        if (process.env.NODE_ENV !== 'production') {
-          if (name.includes('-')) {
-            throw new Error(
-              `URL pattern names should use underscores instead of dashes. Try "${name.replace('-', '_')}"`,
-            );
-          }
+      if (process.env.NODE_ENV !== 'production') {
+        if (name.includes('-')) {
+          throw new Error(
+            `URL pattern names should use underscores instead of dashes. Try "${name.replace('-', '_')}"`,
+          );
         }
-        throw new Error(`URL pattern "${name}" not found`);
-      };
-    } else {
-      urlFunc = this._createUrlFunction(name, patterns);
+      }
+      return;
     }
+    urlFunc = this._createUrlFunction(name, patterns);
 
     // Cache the function
     this._functionCache.set(name, urlFunc);

